src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -541,6 +542,19 @@ abstract class ManagedStrategy(
         if (currentThreadId < 0) return
         // scenario threads are handled separately
         if (currentThreadId < scenario.nThreads) return
+
+        val methodDescriptor = getAsmMethod("void run()").descriptor
+        addBeforeMethodCallTracePoint(
+            owner = runner.testInstance,
+            className = "kotlin.concurrent.Thread",
+            methodName = "run",
+            codeLocation = UNKNOWN_CODE_LOCATION,
+            methodId = MethodIds.getMethodId("kotlin.concurrent.Thread", "run", methodDescriptor),
```

Comment:
I think it is better to set `java.lang.Thread` here instead of `kotlin.concurrent.Thread`.